### ISSUE TITLE
Steering wheel calibration doesn't stay the same after restarting

### ISSUE THREAD

**Issue by Dinir** [Ryochan7/DS4Windows#2681]
**Describe the bug**
The range of 360 Steering Wheel does not stay the same after restarting the app, whether by restarting the entire app or clicking the Stop/Start button. The degree needed to reach the end of the input increases than what it was when the calibration is done.

**To Reproduce**
1. Connect the controller, edit the profile that's being used.
2. Go to Gyro tab, and click `Calibrate` in *360 Steering Wheel - Steering wheel range*.
3. Exit the profile edit window by clicking Save on the top of the window.
4. Stop and Start the app by clicking the corresponding button at the bottom right of the window.
5. Reconnect the controller.

**Expected behavior**
The controller input should be able to reach to the both ends when the controller is rotated by the same degree it needed to reach the ends when the calibration is just done.

**Screenshots and Logs**
I recorded [a video](https://streamable.com/qnakrf) to show how it looks on Controller Readings tab. [Attached log file](https://github.com/Ryochan7/DS4Windows/files/9947103/ds4windows_log.txt) is made during this record. In the video I do the following:

- start DS4Windows, connect the controller
- show the full arc the controller makes
- calibrate the wheel
- show how far the input reaches when the controller rotates by 90 degrees
- stop and start the DS4Windows, reconnect the controller
- show how the input doesn't reach to the both ends after 90 degrees
- show when the input reaches to the both ends

**Desktop (please complete the following information):**
 - Controller Make and Model: Sony DS4 v.2. HID 054c-09cc, manufactured in 2017.07.
 - OS: Windows 10 Pro x64 10.0.19044.0
 - DS4Windows Version: 3.1.9.  
I don't remember the first version I started trying this, but the date is September 29th. I believe new versions released after the date, including 3.1.7, are tested.

**Additional context**
I use 360 Steering Wheel feature to play ETS2. I usually hold the controller mid-air the whole time I play this game, but it's obviously bad for my wrist so I tried to make use of a phone stand to hold it.

I have no problem calibrating it for Steering wheel range of 180. It's not perfectly staying on a plane while calibrating, but that was same when I was holding it mid-air.

Problem occurs when I restart the app. Calibration values stay properly in `ControllerConfigs.xml`, but when I rotate the gamepad 90 degrees to one side, the input only goes to about 80% or less. I try to rotate even further, and sometimes the input reaches to one end when I rotate more than 90 degrees, sometimes it never reaches any of the both ends in any angle during a full revolution.

**Comment by Dinir** [comment]
With the calibration applied, I started getting crashes few minutes after starting the app. Happened twice with the exact same stack trace and it contains `DS4Windows.Mapping.CalculateControllerAngle` and `DS4Windows.Mapping.Scale360degreeGyroAxis`, so I am attaching [one of the log](https://github.com/Ryochan7/DS4Windows/files/9956682/ds4windows_log_20221108.1.txt) here.

**Comment by Ryochan7** [comment]
Ran into this same problem by accident while testing the anti-deadzone behavior for 360 Gyro Steering. Too bad I did not capture the input value that caused the problem. Fails here.

https://github.com/Ryochan7/DS4Windows/blob/master/DS4Windows/DS4Control/Mapping.cs#L6230

```
double angle = Math.Acos(dotProduct / (magAB * magCD));
result = Convert.ToInt32(Global.Clamp(
        -180.0 * C_WHEEL_ANGLE_PRECISION,
        Math.Round((angle * (180.0 / Math.PI)), 1) * C_WHEEL_ANGLE_PRECISION,
        180.0 * C_WHEEL_ANGLE_PRECISION)
     );
```

Going to assume the clamp function succeeded and the failure occurs with the Convert.ToInt32 method.

Other than that, you will likely have to re-calibrate every once in a while but I would not think you would have to do it every single app launch. Calibration has been fine for me once it has been set. Although, I have not kept a calibration for too long before having to test something and wipe the old values anyway.
